paasta_tools/flink_tools.py

```diff
@@ -361,3 +409,443 @@ def get_flink_overview_from_paasta_api_client(
+    )
+
+    return output
+
+
+def collect_flink_job_details(
+    status: Mapping[str, Any],
+    overview: Optional[FlinkClusterOverview],
+    jobs: List[FlinkJobDetails],
+) -> FlinkJobDetailsDict:
+    """Collect job, pod, and resource information.
+
+    :param status: Status dict from flink CR containing state and pod_status
+    :param overview: Flink cluster overview (or None if not running)
+    :param jobs: List of Flink job details
+    :returns: Dictionary containing:
+        - state: Cluster state string
+        - pod_counts: Dict with running, evicted, other, total pod counts
+        - job_counts: Dict with running, finished, failed, cancelled, total job counts (or None)
+        - taskmanagers: Number of taskmanagers (or None)
+        - slots_available: Number of available slots (or None)
+        - slots_total: Total number of slots (or None)
+        - jobs: List of FlinkJobDetails
+    """
+    # Collect pod counts
+    pod_running_count = 0
+    pod_evicted_count = 0
+    pod_other_count = 0
+
+    for pod in status.get("pod_status", []):
+        if pod["phase"] == "Running":
+            pod_running_count += 1
+        elif pod["phase"] == "Failed" and pod.get("reason") == "Evicted":
+            pod_evicted_count += 1
+        else:
+            pod_other_count += 1
+
+    pods_total_count = pod_running_count + pod_evicted_count + pod_other_count
+
+    pod_counts: PodCounts = {
+        "running": pod_running_count,
+        "evicted": pod_evicted_count,
+        "other": pod_other_count,
+        "total": pods_total_count,
+    }
+
+    # Collect job counts if overview is available
+    job_counts: Optional[JobCounts] = None
+    taskmanagers = None
+    slots_available = None
+    slots_total = None
+
+    if overview:
+        jobs_total_count = (
+            overview.jobs_running
+            + overview.jobs_finished
+            + overview.jobs_failed
+            + overview.jobs_cancelled
+        )
+        job_counts = {
+            "running": overview.jobs_running,
+            "finished": overview.jobs_finished,
+            "failed": overview.jobs_failed,
+            "cancelled": overview.jobs_cancelled,
+            "total": jobs_total_count,
+        }
+        taskmanagers = overview.taskmanagers
+        slots_available = overview.slots_available
+        slots_total = overview.slots_total
+
+    return {
+        "state": status["state"],
+        "pod_counts": pod_counts,
+        "job_counts": job_counts,
+        "taskmanagers": taskmanagers,
+        "slots_available": slots_available,
+        "slots_total": slots_total,
+        "jobs": jobs,
+    }
+
+
+def format_flink_state_and_pods(
+    state: str,
+    pod_counts: PodCounts,
+    job_counts: Optional[JobCounts],
+    taskmanagers: Optional[int],
+    slots_available: Optional[int],
+    slots_total: Optional[int],
```

Comment:
hmm, `job_details` is already a TypedDict - would we lose anything if we just passed `job_details` as a whole rather than each attribute individually?